pkg/tfgen/namecheck.go

```diff
@@ -84,26 +84,44 @@ func nameCheck(
 		all = append(all, ps)
 	}
 
+	// It is not required that all resources/functions are mapped in `prov`. We need
```

Comment:
Are we sure his is factually correct? What happens with unmapped resources, don't they result in an error from tfgen and/or have no tokens allocated, or are not present in the schema? Directionally it makes sense though that we should not require all resources to be mapped.

Reply:
Unmapped errors result in an error from tfgen... by default. There are settings that turn off unmapped resource errors, so the rest of the process needs to be resilient to missing resources/functions.

Reply:
Ah that's right. Well that makes sense!